src/redteam/extraction/purpose.ts

```diff
@@ -1,11 +1,44 @@
 import dedent from 'dedent';
+import { fetchWithCache } from '../../cache';
+import { getEnvBool } from '../../envars';
+import logger from '../../logger';
+import { REQUEST_TIMEOUT_MS } from '../../providers/shared';
 import type { ApiProvider } from '../../types';
+import { RedTeamGenerationResponse, REMOTE_GENERATION_URL } from './common';
 import { callExtraction, formatPrompts } from './util';
 
 export async function extractSystemPurpose(
   provider: ApiProvider,
   prompts: string[],
 ): Promise<string> {
+  const useRemoteGeneration = !getEnvBool('PROMPTFOO_DISABLE_REDTEAM_REMOTE_GENERATION', false);
+
+  if (useRemoteGeneration) {
+    try {
+      const body = {
+        task: 'purpose',
+        prompts: prompts,
+      };
+
+      const response = await fetchWithCache(
```

Comment:
so much nicer than the provider approach!

shall we put `fetchRemoteGeneration` in `common`?  Otherwise you're importing the pieces (`RedTeamGenerationResponse` and `REMOTE_GENERATION_URL`) and constructing the request multiple times.